src/Array.ts

```diff
@@ -0,0 +1,139 @@
+import { clip } from "./Number";
+
+
+export function getValue<T>(array: T[], index: number): T {
+    return array[index];
+}
+
+
+export function setValue<T>(array: T[], index: number, value: T): void {
+    array[index] = value;
+}
+
+
+export function at<T>(array: T[], index: number): T {
+    return 0 <= index ? array[index] : array[array.length + index - 1];
+}
+
+
+export function lastAt<T>(array: T[], index: number): T {
+    return array[array.length - index - 1];
+}
+
+
+export function first<T>(array: T[]): T {
+    return array[0];
+}
+
+
+export function setFirst<T>(array: T[], value: T): void {
+    array[0] = value;
+}
+
+
+export function last<T>(array: T[]): T {
+    return array[array.length - 1];
+}
+
+
+export function setLast<T>(array: T[], value: T): void {
+    array[array.length - 1] = value;
+}
+
+
+export function copy<T>(source: T[], destination: T[]): void;
+export function copy<T>(source: T[], destination: T[], length: number): void;
+export function copy<T>(
+    source: T[], sourceOffset: number, destination: T[], destinationOffset: number, length: number
+): void;
+export function copy<T>(src: T[], ...args: T[]): void {
+    let srcOffset: number = 0;
+    let dst: T[];
+    let dstOffset: number = 0;
+    let len: number = source.length;
+
+    switch (args.length) {
+    case 1:
+        [dst] = args;
+        break;
+    case 2:
+        [dst, len] = args;
+        break;
+    case 4:
+        [srcOffset, dst, dstOffset, len] = args;
+        break;
+    default:
+        throw new Error("Invalid overload arguments");
+    } 
```

Comment:
Codacy has a fix for the issue: [Trailing spaces not allowed.](https://app.codacy.com/gh/simpler-one/node.utils/pullRequest?prid=7629571)
```suggestion
    }
```